### PR TITLE
Reconnect when websocket unexpectedly closes

### PR DESCRIPTION
**Pull request by robertlong**
Todo:
- [x] Merge and update update minijanus version
- [x] Update networked-aframe to allow `.connect()` to return a promise: https://github.com/networked-aframe/networked-aframe/pull/125
- [x] Add a maximum number of reconnects.
- [x] Determine if reconnection jitter/backoff strategy is correct.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
   },
   "dependencies": {
     "debug": "^3.1.0",
-    "minijanus": "^0.5.0"
+    "minijanus": "https://github.com/mozilla/minijanus.js#fix/destroy-session"
```
Comment (brianpeiris) [inline package.json]: Please publish minijanus to npm and use a version number here.

File: src/index.js [hunk 2]
```diff
@@ -135,6 +181,48 @@ class JanusAdapter {
     await Promise.all(this.publisher.initialOccupants.map(this.addOccupant.bind(this)));
   }
 
+  onWebsocketClose(event) {
+    // The connection was closed successfully. Don't try to reconnect.
+    if (event.code === 1000) {
```
Comment (brianpeiris) [inline src/index.js]: Maybe put thus number in a named constant?

File: src/index.js [hunk 3]
```diff
@@ -486,23 +578,31 @@ class JanusAdapter {
...
+    if (this.publisher) {
+      this.publisher.unreliableChannel.send(JSON.stringify({ dataType, data }));
+    }
   }
 
   broadcastDataGuaranteed(dataType, data) {
-    this.publisher.reliableChannel.send(JSON.stringify({ dataType, data }));
+    if (this.publisher) {
+      this.publisher.reliableChannel.send(JSON.stringify({ dataType, data }));
+    }
```
Comment (brianpeiris) [inline src/index.js]: Hmm, it's a bit troubling if something tries to send data before a publisher is established. Maybe add a console warning in an else statement in these methods?

Comment (robertlong) [reply src/index.js]: Yeah, I'll add that. I don't like that we have to do this check at all. It feels bad that a call to `broadcastDataGuaranteed` is not guaranteed to broadcast data or error. Unfortunately I'm not sure how to fix this without making larger changes to networked-aframe to add the concept of an adapter that is in a reconnecting state and avoid sending data altogether. I don't think I want to tackle that just yet.

**Pull request merged**